pymeasure/instruments/signalrecovery/dsp7265.py

```diff
@@ -556,41 +558,29 @@ def buffer_to_float(self, buffer_data, sensitivity=None, sensitivity2=None, rais
                     int(format(v2, "016b") + format(v1, "016b"), 2) / 1000 for
                     v1, v2 in zip(buffer_data["frequency part 1"], buffer_data["frequency part 2"])
                 ])
-            elif raise_error:
-                raise ValueError("Cannot calculate the frequency when not both"
-                                 "frequency part 1 and 2 are provided.")
+            else:
+                maybe_raise("Can calculate the frequency only when both"
+                            "frequency part 1 and 2 are provided.")
 
         # conversion for, adc1, adc2, dac1, dac2, ratio, and log ratio
-        for key in ["adc1", "adc2", "dac1", "dac2", "ratio", "log ratio"]:
-            if key in buffer_data:
-                data[key] = buffer_data[key] / 1000
+        convert_if_present(["adc1", "adc2", "dac1", "dac2", "ratio", "log ratio"], 1 / 1000)
 
         # adc3 (integrating converter); requires a call to adc3_time
-        if "adc3" in buffer_data:
-            data["adc3"] = buffer_data["adc3"] / (50000 * self.adc3_time)
+        convert_if_present(["adc3"], 1 / (50000 * self.adc3_time))
```

Comment:
Downside of this approach is that a call to adc3_time (and hence communication with the instrument) is now always executed, even if adc3 not present.

Reply:
Good point, I undid that.

Reply:
Yes, I will!

Reply:
> Good point, I undid that. Maybe when you are with the instrument again you can do a quick check?

I just did a test with the instrument and as far as I can tell right now it works perfectly fine.